File: ds assignment 3/q2 assignment 3.py
def reverse_string(s):
    stack = []
    for char in s:
        stack.append(char)

    reversed_str = ""
    while stack:
        reversed_str += stack.pop()
    return reversed_str


# Driver Code
string = input("Enter a string: ")
print("Reversed String:", reverse_string(string))

File: ds assignment 3/q2 assignment 3.cpp
// Assignment 3 Q2.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

string reverseString(string str) {
    stack<char> s;
    for (char ch : str) s.push(ch);
    string rev = "";
    while (!s.empty()) {
        rev += s.top();
        s.pop();
    }
    return rev;
}

int main() {
    string input;
    cout << "Enter a string: ";
    cin >> input;
    cout << "Reversed string: " << reverseString(input) << endl;
    return 0;
}
